src/python.rs

```diff
@@ -59,6 +59,10 @@ struct PythonParser<'j> {
 impl<'j> PythonParser<'j> {
     fn py_take_value<StringCache: StringMaybeCache>(&mut self, py: Python, peek: Peek) -> JsonResult<PyObject> {
         match peek {
+            Peek::Null => {
+                self.parser.consume_null()?;
+                Ok(py.None())
+            }
```

Comment:
For what it's worth the compiler will probably reorder branches in a match for what it thinks makes most sense anyway (as long as they're not gated by conditionals, only one of these branches will be traversed), so moving this likely does nothing.